questionnaire/dataset/dataset-routes.js

```diff
@@ -17,6 +17,7 @@ router
 
             const resourceCollection = await datasetService.getResource(
                 questionnaireId,
+                req.header('On-behalf-of'),
```

Comment:
the header name is sentence case here, whereas it is all lowercase in the spec. Can we make that consistent, which ever is chosen